lib/membrane_file/source.ex

```diff
@@ -17,17 +17,27 @@ defmodule Membrane.File.Source do
                 spec: pos_integer(),
                 default: 2048,
                 description: "Size of chunks being read"
+              ],
+              content_format: [
+                spec: module() | nil,
+                default: nil,
+                description: "Caps which will be sent from Source"
               ]
 
   def_output_pad :output, caps: {RemoteStream, type: :bytestream}
 
   @impl true
-  def handle_init(%__MODULE__{location: location, chunk_size: size}) do
+  def handle_init(%__MODULE__{
+        location: location,
+        chunk_size: size,
+        content_format: content_format
+      }) do
```

Comment:
This started to grow too much. I guess you can just do

```suggestion
  def handle_init(options) do
    state =
      options
      |> Map.from_struct()
      |> Map.update(:location, &Path.expand/1)
      
    {:ok, state}
  end
```